### PR TITLE
Do not scan archived volumes for external subs or audio.

### PR DESCRIPTION
**Pull request by ace20022**
This pr superseeds #9929

### CODE REVIEW

File: xbmc/Util.cpp [hunk 1]
```diff
@@ -2313,10 +2319,26 @@ void CUtil::ScanForExternalAudio(const std::string& videoPath, std::vector<std::
...
+  if (URIUtils::IsInRAR(videoPath))
+    ExcludeMovieRar(videoPath, items);
+
   std::vector<std::string> exts = StringUtils::Split(CServiceBroker::GetFileExtensionProvider().GetMusicExtensions(), "|");
   ScanPathsForAssociatedItems(strAudio, items, exts, vecAudio);
 }
 
+void CUtil::ExcludeMovieRar(const std::string& strMovie, CFileItemList& items)
+{
+  std::string strRarFileName = URIUtils::ReplaceExtension(CURL(URIUtils::GetRealPath(strMovie)).GetHostName(), "");
```
Comment (garbear) [inline xbmc/Util.cpp]: This could be split into several lines to improve readability on small screens

File: xbmc/Util.cpp [hunk 2]
```diff
@@ -2313,10 +2319,26 @@ void CUtil::ScanForExternalAudio(const std::string& videoPath, std::vector<std::
...
+  if (StringUtils::StartsWith(URIUtils::GetExtension(strRarFileName), ".part"))
+    strRarFileName = URIUtils::ReplaceExtension(strRarFileName, "") + ".part";
+  else
+    strRarFileName += ".r";
+  for (int i = items.Size() - 1; i > 0; i--){
```
Comment (garbear) [inline xbmc/Util.cpp]: brace on separate line

File: xbmc/Util.cpp [hunk 3]
```diff
@@ -2006,7 +2006,10 @@ void CUtil::ScanForExternalSubtitles(const std::string& strMovie, std::vector<st
   const std::vector<std::string> common_sub_dirs = { "subs", "subtitles", "vobsubs", "sub", "vobsub", "subtitle" };
   GetItemsToScan(strBasePath, CServiceBroker::GetFileExtensionProvider().GetSubtitleExtensions(), common_sub_dirs, items);
 
-  if (!CMediaSettings::GetInstance().GetAdditionalSubtitleDirectoryChecked() && !CServiceBroker::GetSettings().GetString(CSettings::SETTING_SUBTITLES_CUSTOMPATH).empty()) // to avoid checking non-existent directories (network) every time..
+  if (URIUtils::IsInRAR(strMovie))
+    ExcludeMovieRar(strMovie, items);
+
+  if (!CMediaSettings::GetInstance().GetAdditionalSubtitleDirectoryChecked() && !CSettings::GetInstance().GetString(CSettings::SETTING_SUBTITLES_CUSTOMPATH).empty()) // to avoid checking non-existent directories (network) every time..
```
Comment (Rechi) [inline xbmc/Util.cpp]: `!CServiceBroker::GetSettings().GetString(...)`

**Comment by notspiff** [comment]
i have no further comments @ace20022

**Comment by Rechi** [comment]
Is this ready to get merged?

**Comment by ace20022** [comment]
No.

**Comment by phunkyfish** [comment]
PR closed as it appears to be abandoned by the author. 

If the author would like it kept open please just request so on the PR. If you feel there is still merit in merging this PR you can request the label be changed to `PR Cleanup: Backburner`.

**Pull request closed without merging**